lib/github/markup.rb

```diff
@@ -25,17 +43,34 @@ def render(filename, content = nil)
         content
       end
     end
-
-    def markup(file, pattern, opts = {}, &block)
-      markups << GemImplementation.new(pattern, file, &block)
+    
+    def render_s(symbol, content)
+      if content.nil?
+        raise ArgumentError, 'Can not render a nil.'
+      elsif markups.has_key?(symbol)
+        markups[symbol].render(content)
+      else
+        content
+      end
+    end
+    
+    def markup(symbol, file, pattern, opts = {}, &block)
+      markup_impl(symbol, GemImplementation.new(pattern, file, &block))
+    end
+    
+    def markup_impl(symbol, impl)
+      if markups.has_key?(symbol)
+        raise ArgumentError, "The '#{symbol}' symbol is already defined."
+      end
+      markups[symbol] = impl
     end
 
-    def command(command, regexp, name, &block)
+    def command(symbol, command, regexp, name, &block)
```

Comment:
I'm not sure I like the idea of changing an existing method signature for this feature. Can this be handled in a different method (or a different way altogether?).